_posts/2019-10-27-on-code-reviews.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: post
+permalink: on-code-reviews
+title: 'On Code Reviews'
+tags: [programming]
+---
+
+I wanted to share some insights into how I review code, learning from others,
+helping others, and getting a better overview of the code base which ultimately
+allows me to write better[^1] code. This post is inspired by a recent
+[tweet](https://twitter.com/kristoferbaxter/status/1186290473322741760), a]
+similar [blog post](https://techbeacon.com/app-dev-testing/10-commandments-navigating-code-reviews)
+(by [Angie Jones](https://twitter.com/techgirl1908)) and a [talk](https://www.youtube.com/watch?v=Ea8EiIPZvh0)
+on the topic (by [April Wensel](https://twitter.com/aprilwensel)) from 2018's
+try! Swift NYC.
+
+- **I point out the good things.** This goes for anything really; not just code
+reviews. If there's something you like or learned, let the author know. I
+believe this will make a big difference on how a review is perceived, and is a
+win-win all around.<br />
+This is what I believe to be the **single most important part of a review**.
+- **I read the pull request's description _and_ ticket.** Yes, also the ticket.
+Seldom does that not help me getting a better understanding of the code. And
+understanding is key. This makes everyone involved's time a lot more valuable.
+If something in the description or ticket is unclear to me, I try to reach out
+the the author before continuing a review.
+- **I provide context.** If I come across something that "might better be
+represented by a function rather than a computed property", I explain why I
+would suggest that in the given case.
+- **I provide references.** Whether it's a blog post, piece of code, or
+documentation I reference, I link it. This allows the author to dig deeper into
+the topic if they want, and may be helpful for those who read the comment(s) in
+the future.
+- **I take the time to try and understand the code.** When I feel like I'm
+having a hard time grasping some of the decisions that have been made, or the
+code that has been written in this pull request, I often feel asking about it
+in writing is difficult. If this happens, I either take a step away from the
+computer and try again later, or ask to go over it in person (face-to-face or
+in a video call, _not_ a phone call. Being able to see each other aids
```

Comment:
```suggestion
in a video call, _not_ a phone call, nor email or even Slack). Being able to see each other aids
```

Reply:
I'm going to skip on these as I'm assuming anything "non-voice" is a no-go here anyway. 😊